src/main/java/uk/gov/hmcts/ccd/service/MessagePublisherRunnable.java

```diff
@@ -63,10 +65,31 @@ private void processUnpublishedMessages(Pageable pageable,
         }
     }
 
+    private String getProperty(JsonNode data, String header) {
+        try {
+            return data.get(header).asText();
+        } catch (NullPointerException e) {
+            return "null";
+        }
+    }
+
     private void publishMessages(Slice<MessageQueueCandidateEntity> messagesToPublish,
                                  List<MessageQueueCandidateEntity> processedEntities) {
         messagesToPublish.get().forEach(entity -> {
-            jmsTemplate.convertAndSend(publishMessageTask.getDestination(), entity.getMessageInformation());
+            jmsTemplate.convertAndSend(
+                publishMessageTask.getDestination(),
+                entity.getMessageInformation(),
+                messagePostProcessor -> {
+                    for (String property : properties) {
+                        if (!(getProperty(entity.getMessageInformation(), property).equals("null"))) {
+                            messagePostProcessor.setStringProperty(property,
+                                getProperty(entity.getMessageInformation(), property)
```

Comment:
this will only work if the property has got the same name as the field in the message...it's very likely that in the future we'll have a property where that's not the case. Can we find now a better implementation to cope with that possible case?